segfast/utils.py

```diff
@@ -8,11 +8,26 @@
 except ImportError:
     try:
         from tqdm.auto import tqdm
-        def Notifier(pbar, *args, **kwargs):
-            """ Progress bar. """
-            if pbar:
-                return tqdm(*args, **kwargs)
-            return lambda iterator: iterator
+
+        class Notifier:
+            """ tqdm notifier. """
+            def __init__(self, pbar, total, *args, **kwargs):
+                self.pbar = pbar
+                self.total = total
+                self.args = args
+                self.kwargs = kwargs
+
+            def __call__(self, iterator, *args, **kwargs):
+                _ = args, kwargs
+                return tqdm(iterator, total=self.total, disable=not self.pbar)
+
+            def __enter__(self):
+                """ !!. """
+                return tqdm(*self.args, total=self.total,**self.kwargs, disable=not self.pbar)
```

Comment:
Just to unify `bar` -- `pbar` and `__call__` with notifier from `batchflow` and (partially) avoid the case when an iterable can be passed to `__init__`, but actual iteration in `__call__` is performed over another object.

```suggestion
        from functools import partial  # insert before import from tqdm

        class Notifier:
            """ tqdm notifier. """
            def __init__(self, bar=False, total=None, **kwargs):
                self.pbar = partial(tqdm, disable=not bar, total=total, **kwargs)

            def __call__(self, iterable):
                return self.pbar(iterator, total=self.total, disable=not self.pbar)

            def __enter__(self):
                """ !!. """
                return self.pbar()
```